src/setup-haskell.ts

```diff
@@ -26,9 +26,15 @@ async function cabalConfig(): Promise<string> {
       );
 
     if (opts.stack.setup)
-      await core.group('Pre-installing GHC with stack', async () =>
-        exec('stack', ['setup', opts.ghc.resolved])
-      );
+      await core.group('Pre-installing GHC with stack', async () => {
+        await exec('stack', ['setup', opts.ghc.resolved]);
```

Comment:
This codepath isn't executed every time stack is installed. The action first checks to see if the tool already exists, and if so, exits early. The `success` function in `src/installer.ts` is responsible for setting this up correctly and is probably the best place to put this logic.

I'm glad this was submitted, though! Walking through the code helped me realize that I had placed some stack specific code beyond the early exit codepath, making it (probably) possible to end up with a windows system without `STACK_ROOT` set correctly.